src/v2/Components/FlashBanner/index.tsx

```diff
@@ -1,36 +1,65 @@
 import React from "react"
 import qs from "qs"
 import { Banner, Sans } from "@artsy/palette"
+import { QueryRenderer, graphql } from "react-relay"
+import { useSystemContext } from "v2/Artsy"
+
+type Message = string | React.ComponentType
 
 /** Flash messages triggered by a url `flash_message` query param
  * A message can be a string or a React Component that takes no props
  * (that is, use relay or context if necessary)
  */
-const messageByQueryParam: { [k: string]: string | React.FC } = {
+const messageByCode: { [k: string]: Message } = {
   confirmed: "Your email has been confirmed.",
   already_confirmed: "You have already confirmed your email.",
   invalid_token: "An error has occurred. Please contact support@artsy.net.",
   blank_token: "An error has occurred. Please contact support@artsy.net.",
   expired_token: "Link expired. Resend verification email.",
 }
 
+const priorityCodes = ["expired_token", "blank_token", "invalid_token"]
+
+interface Props {
+  messageCode?: string
+  me?: {
+    canRequestEmailConfirmation: boolean
+  }
+}
+
 /**
  * The component responsible for selecting a flash message key from the `flash_message` url query param
  */
-export const FlashBanner: React.FunctionComponent<{
-  messageCode?: string
-}> = props => {
-  const [messageCode, setMessageCode] = React.useState(props.messageCode)
+export const BaseFlashBanner: React.FunctionComponent<Props> = props => {
+  const [Message, setMessage] = React.useState<Message>(
+    props.messageCode ? messageByCode[props.messageCode] : null
+  )
 
   React.useEffect(() => {
-    if (!messageCode) {
+    if (!Message) {
+      let ShowMessage: Message = null
       const query = qs.parse(window.location.search.slice(1))
-      if (query["flash_message"]) setMessageCode(query["flash_message"])
+      const messageCode = query["flash_message"]
+
+      // Prompt the user to verify email if they can
+      if (props.me?.canRequestEmailConfirmation) {
+        ShowMessage = "You should really do email verification"
+      }
+
+      // Display a message based on url flash_message param, ignoring
+      // the email verification prompt if it should be overriden by a
+      // priority message
+      if (
+        messageCode &&
+        (!ShowMessage || priorityCodes.includes(messageCode))
```

Comment:
Having multiple lists feels a bit confusing, `priorityCode` and `messageByCode` -- why not one object and one lookup? Then everything can be contained in one place, with comments added or keys ordered by logical grouping.